### ISSUE TITLE
Fix php docblocks tags

### ISSUE THREAD

**Issue by benlk** [INN/Largo#321]
@param int $example This is an example function/method parameter description.

http://www.phpdoc.org/docs/latest/references/phpdoc/basic-syntax.html

http://www.phpdoc.org/docs/latest/references/phpdoc/tags/param.html

**Comment by aschweigert** [comment]
Let's also check @since in the doc blocks because a lot of them say 1.0 and should really be either 0.3 or 0.4. That is, generally, let's try to fix all the doc blocks and make sure they're as complete as possible, formatted correctly, etc.

**Comment by aschweigert** [comment]
OK, so instead of keeping this in one giant issue, let's split it up.

Seems to me we should have one issue per file and the issue should have two parts:

1) make sure the file has a header block that explains what's going on, acts as a table of contents (for larger files), etc. 

2) review the php docblocks and other inline documentation and make sure everything is formatted consistently (let's make sure our code style guide covers this before we start this process - https://github.com/INN/docs/tree/master/style-guides/code)

Anything else? @willhaynes you want to take the first crack at this? or maybe @tothebeat?

**Comment by tothebeat** [comment]
I've been looking for a way to programmatically audit our code and find which files and functions therein need documentation. With such an audit, it would be straightforward to create tickets for each file and show what's missing from each.

I went down a rabbit hole with phpdoc to show undocumented elements, and apparently its `undocumentedelements` option doesn't actually work. It's in the [phpdoc documentation](https://pear.php.net/reference/PhpDocumentor-latest/phpDocumentor/tutorial_phpDocumentor.howto.pkg.html#using.command-line.undocumentedelements) and the process is explained in [this stackoverflow answer](http://stackoverflow.com/a/1936421/4677160), however actually trying to use this switch (`phpdoc --filename functions.php --target doc --undocumentedelements`) results in a RuntimeException and the error `The "--undocumentedelements" option does not exist.`.

@willhaynes has done some good work in #333 on using doxphp to generate documentation suitable for use on readthedocs. He demonstrated that the documentation generated from `inc/images.php`  on [readthedocs](http://largo.readthedocs.org/api/inc/images.html) has a stub for every function and clearly shows when a function does not have any inline documentation.

**Comment by benlk** [comment]
Does `-ue` work? It's an alternate flag for `--undocumentedelements`.

**Comment by willhaynes** [comment]
@tothebeat If we generated a list of files we care about, we could push all current documentation for those to readthedocs. That might be an easier way to see what has documentation and what doesn't.

**Comment by drywall** [comment]
I'm not an expert but I think phpcs (PHP_CodeSniffer) should be able to do this with the right ruleset. See e.g. http://stackoverflow.com/questions/13767550/is-there-a-phpcs-standard-targeting-php-docblocks

**Comment by tothebeat** [comment]
@benlk I tried `-ue` as well and got an error that `The "-u" option does not exist.`, so it's not even parsing that option properly.. I'm using phpDocumentor version 2.8.2 that I [installed via PEAR](http://www.phpdoc.org/docs/latest/getting-started/installing.html).

@willhaynes I think that's a good idea. The question I then have is what qualifies a file as one we should care about? Or, what would qualify a file as one we don't care about?

**Comment by benlk** [comment]
:+1: on generating for all the PHP files, because we can get more eyes looking at the undocumented files.

**Comment by tothebeat** [comment]
As of 209313206c7fdb123127a8cfa2ba5bb0644586e0 there is now automatically generated documentation for each PHP file in the project on readthedocs. For example, the file [inc/ajax-functions.php](https://github.com/INN/Largo/blob/develop/inc/ajax-functions.php) on the develop branch will appear at http://largo.readthedocs.org/en/develop/api/inc/ajax-functions.html.

For a file `/path/to/filename.php` from the root of the Largo project, the corresponding documentation will appear in `http://largo.readthedocs.org/en/develop/api/path/to/filename.html`.

Caveats:

- As it exists now, the docs are being generated for a static set of PHP files. If a PHP file is added to the project then a corresponding line will have to be added to the [Makefile](https://github.com/INN/Largo/blob/develop/docs/Makefile#L179).

- There is no automatic index created for the files on readthedocs. Unless explicitly linked, there is no way to know that documentation exists for a given PHP file on readthedocs.

- All non-library PHP files in the project are having documentation automatically generated for them. If it doesn't make sense for a file to have API documentation, it needs to be removed from the [Makefile](https://github.com/INN/Largo/blob/develop/docs/Makefile#L179).

- The documentation is not automatically updated on changes to the PHP files. If you are using Grunt, you need to run the command `grunt apidocs`. This assumes that Grunt and prerequisites are installed, along with [doxphp](https://github.com/avalanche123/doxphp) (which includes doxphp2sphinx).

**Comment by tothebeat** [comment]
This is a listing of all PHP files that are included in the [Makefile](https://github.com/INN/Largo/blob/develop/docs/Makefile#L179) for inline code documentation extraction. Each item links to the source on the `develop` branch, and includes a link to the corresponding documentation page on Read The Docs.

If a file doesn't belong here, edit this comment and move the item from the "Needs Documentation" section to the "Will Not Document" section. Please indicate in a note after the item why it will not be documented. I suspect that if any are removed, that an entire class of files will be removed instead of one by one (template files, for example). Once we determine what belongs and doesn't belong, I'll clean up what's included in the Makefile.

## Needs Documentation

- [ ] [./404.php](https://github.com/INN/Largo/blob/develop/404.php) - [docs](http://largo.readthedocs.org/en/develop/api/404.html)
- [ ] [./archive.php](https://github.com/INN/Largo/blob/develop/archive.php) - [docs](http://largo.readthedocs.org/en/develop/api/archive.html)
- [ ] [./category.php](https://github.com/INN/Largo/blob/develop/category.php) - [docs](http://largo.readthedocs.org/en/develop/api/category.html)
- [ ] [./comments.php](https://github.com/INN/Largo/blob/develop/comments.php) - [docs](http://largo.readthedocs.org/en/develop/api/comments.html)
- [ ] [./feed-mailchimp.php](https://github.com/INN/Largo/blob/develop/feed-mailchimp.php) - [docs](http://largo.readthedocs.org/en/develop/api/feed-mailchimp.html)
- [ ] [./footer.php](https://github.com/INN/Largo/blob/develop/footer.php) - [docs](http://largo.readthedocs.org/en/develop/api/footer.html)
- [ ] [./full-page.php](https://github.com/INN/Largo/blob/develop/full-page.php) - [docs](http://largo.readthedocs.org/en/develop/api/full-page.html)
- [ ] [./functions.php](https://github.com/INN/Largo/blob/develop/functions.php) - [docs](http://largo.readthedocs.org/en/develop/api/functions.html)
- [ ] [./header.php](https://github.com/INN/Largo/blob/develop/header.php) - [docs](http://largo.readthedocs.org/en/develop/api/header.html)
- [ ] [./home.php](https://github.com/INN/Largo/blob/develop/home.php) - [docs](http://largo.readthedocs.org/en/develop/api/home.html)
- [ ] [./image.php](https://github.com/INN/Largo/blob/develop/image.php) - [docs](http://largo.readthedocs.org/en/develop/api/image.html)
- [ ] [./index.php](https://github.com/INN/Largo/blob/develop/index.php) - [docs](http://largo.readthedocs.org/en/develop/api/index.html)
- [ ] [./largo-apis.php](https://github.com/INN/Largo/blob/develop/largo-apis.php) - [docs](http://largo.readthedocs.org/en/develop/api/largo-apis.html)
- [ ] [./options.php](https://github.com/INN/Largo/blob/develop/options.php) - [docs](http://largo.readthedocs.org/en/develop/api/options.html)
- [ ] [./page.php](https://github.com/INN/Largo/blob/develop/page.php) - [docs](http://largo.readthedocs.org/en/develop/api/page.html)
- [ ] [./search.php](https://github.com/INN/Largo/blob/develop/search.php) - [docs](http://largo.readthedocs.org/en/develop/api/search.html)
- [ ] [./searchform.php](https://github.com/INN/Largo/blob/develop/searchform.php) - [docs](http://largo.readthedocs.org/en/develop/api/searchform.html)
- [ ] [./series-landing.php](https://github.com/INN/Largo/blob/develop/series-landing.php) - [docs](http://largo.readthedocs.org/en/develop/api/series-landing.html)
- [ ] [./sidebar.php](https://github.com/INN/Largo/blob/develop/sidebar.php) - [docs](http://largo.readthedocs.org/en/develop/api/sidebar.html)
- [ ] [./single-one-column.php](https://github.com/INN/Largo/blob/develop/single-one-column.php) - [docs](http://largo.readthedocs.org/en/develop/api/single-one-column.html)
- [ ] [./single-two-column.php](https://github.com/INN/Largo/blob/develop/single-two-column.php) - [docs](http://largo.readthedocs.org/en/develop/api/single-two-column.html)
- [ ] [./single.php](https://github.com/INN/Largo/blob/develop/single.php) - [docs](http://largo.readthedocs.org/en/develop/api/single.html)
- [ ] [./homepages/homepage-class.php](https://github.com/INN/Largo/blob/develop/homepages/homepage-class.php) - [docs](http://largo.readthedocs.org/en/develop/api/homepages/homepage-class.html)
- [ ] [./homepages/homepage.php](https://github.com/INN/Largo/blob/develop/homepages/homepage.php) - [docs](http://largo.readthedocs.org/en/develop/api/homepages/homepage.html)
- [ ] [./homepages/layouts/HomepageBlog.php](https://github.com/INN/Largo/blob/develop/homepages/layouts/HomepageBlog.php) - [docs](http://largo.readthedocs.org/en/develop/api/homepages/layouts/HomepageBlog.html)
- [ ] [./homepages/layouts/HomepageSingle.php](https://github.com/INN/Largo/blob/develop/homepages/layouts/HomepageSingle.php) - [docs](http://largo.readthedocs.org/en/develop/api/homepages/layouts/HomepageSingle.html)
- [ ] [./homepages/layouts/HomepageSingleWithFeatured.php](https://github.com/INN/Largo/blob/develop/homepages/layouts/HomepageSingleWithFeatured.php) - [docs](http://largo.readthedocs.org/en/develop/api/homepages/layouts/HomepageSingleWithFeatured.html)
- [ ] [./homepages/layouts/HomepageSingleWithSeriesStories.php](https://github.com/INN/Largo/blob/develop/homepages/layouts/HomepageSingleWithSeriesStories.php) - [docs](http://largo.readthedocs.org/en/develop/api/homepages/layouts/HomepageSingleWithSeriesStories.html)
- [ ] [./homepages/layouts/LegacyThreeColumn.php](https://github.com/INN/Largo/blob/develop/homepages/layouts/LegacyThreeColumn.php) - [docs](http://largo.readthedocs.org/en/develop/api/homepages/layouts/LegacyThreeColumn.html)
- [ ] [./homepages/layouts/TopStories.php](https://github.com/INN/Largo/blob/develop/homepages/layouts/TopStories.php) - [docs](http://largo.readthedocs.org/en/develop/api/homepages/layouts/TopStories.html)
- [ ] [./homepages/templates/blog.php](https://github.com/INN/Largo/blob/develop/homepages/templates/blog.php) - [docs](http://largo.readthedocs.org/en/develop/api/homepages/templates/blog.html)
- [ ] [./homepages/templates/full-width-image.php](https://github.com/INN/Largo/blob/develop/homepages/templates/full-width-image.php) - [docs](http://largo.readthedocs.org/en/develop/api/homepages/templates/full-width-image.html)
- [ ] [./homepages/templates/legacy-three-column.php](https://github.com/INN/Largo/blob/develop/homepages/templates/legacy-three-column.php) - [docs](http://largo.readthedocs.org/en/develop/api/homepages/templates/legacy-three-column.html)
- [ ] [./homepages/templates/top-stories.php](https://github.com/INN/Largo/blob/develop/homepages/templates/top-stories.php) - [docs](http://largo.readthedocs.org/en/develop/api/homepages/templates/top-stories.html)
- [ ] [./homepages/zones/zones.php](https://github.com/INN/Largo/blob/develop/homepages/zones/zones.php) - [docs](http://largo.readthedocs.org/en/develop/api/homepages/zones/zones.html)
- [ ] [./inc/ad-codes.php](https://github.com/INN/Largo/blob/develop/inc/ad-codes.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/ad-codes.html)
- [ ] [./inc/ajax-functions.php](https://github.com/INN/Largo/blob/develop/inc/ajax-functions.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/ajax-functions.html)
- [ ] [./inc/avatars.php](https://github.com/INN/Largo/blob/develop/inc/avatars.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/avatars.html)
- [ ] [./inc/custom-feeds.php](https://github.com/INN/Largo/blob/develop/inc/custom-feeds.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/custom-feeds.html)
- [ ] [./inc/custom-less-variables.php](https://github.com/INN/Largo/blob/develop/inc/custom-less-variables.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/custom-less-variables.html)
- [ ] [./inc/dashboard.php](https://github.com/INN/Largo/blob/develop/inc/dashboard.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/dashboard.html)
- [ ] [./inc/deprecated.php](https://github.com/INN/Largo/blob/develop/inc/deprecated.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/deprecated.html)
- [ ] [./inc/editor.php](https://github.com/INN/Largo/blob/develop/inc/editor.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/editor.html)
- [ ] [./inc/enqueue.php](https://github.com/INN/Largo/blob/develop/inc/enqueue.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/enqueue.html)
- [ ] [./inc/featured-content.php](https://github.com/INN/Largo/blob/develop/inc/featured-content.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/featured-content.html)
- [ ] [./inc/featured-media.php](https://github.com/INN/Largo/blob/develop/inc/featured-media.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/featured-media.html)
- [ ] [./inc/header-footer.php](https://github.com/INN/Largo/blob/develop/inc/header-footer.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/header-footer.html)
- [ ] [./inc/helpers.php](https://github.com/INN/Largo/blob/develop/inc/helpers.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/helpers.html)
- [ ] [./inc/home-templates.php](https://github.com/INN/Largo/blob/develop/inc/home-templates.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/home-templates.html)
- [ ] [./inc/images.php](https://github.com/INN/Largo/blob/develop/inc/images.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/images.html)
- [ ] [./inc/metabox-api.php](https://github.com/INN/Largo/blob/develop/inc/metabox-api.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/metabox-api.html)
- [ ] [./inc/nav-menus.php](https://github.com/INN/Largo/blob/develop/inc/nav-menus.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/nav-menus.html)
- [ ] [./inc/open-graph.php](https://github.com/INN/Largo/blob/develop/inc/open-graph.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/open-graph.html)
- [ ] [./inc/plugin-init.php](https://github.com/INN/Largo/blob/develop/inc/plugin-init.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/plugin-init.html)
- [ ] [./inc/post-metaboxes.php](https://github.com/INN/Largo/blob/develop/inc/post-metaboxes.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/post-metaboxes.html)
- [ ] [./inc/post-tags.php](https://github.com/INN/Largo/blob/develop/inc/post-tags.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/post-tags.html)
- [ ] [./inc/post-templates.php](https://github.com/INN/Largo/blob/develop/inc/post-templates.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/post-templates.html)
- [ ] [./inc/related-content.php](https://github.com/INN/Largo/blob/develop/inc/related-content.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/related-content.html)
- [ ] [./inc/robots.php](https://github.com/INN/Largo/blob/develop/inc/robots.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/robots.html)
- [ ] [./inc/sidebars.php](https://github.com/INN/Largo/blob/develop/inc/sidebars.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/sidebars.html)
- [ ] [./inc/taxonomies.php](https://github.com/INN/Largo/blob/develop/inc/taxonomies.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/taxonomies.html)
- [ ] [./inc/term-icons.php](https://github.com/INN/Largo/blob/develop/inc/term-icons.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/term-icons.html)
- [ ] [./inc/term-meta.php](https://github.com/INN/Largo/blob/develop/inc/term-meta.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/term-meta.html)
- [ ] [./inc/term-sidebars.php](https://github.com/INN/Largo/blob/develop/inc/term-sidebars.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/term-sidebars.html)
- [ ] [./inc/update.php](https://github.com/INN/Largo/blob/develop/inc/update.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/update.html)
- [ ] [./inc/users.php](https://github.com/INN/Largo/blob/develop/inc/users.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/users.html)
- [ ] [./inc/verify.php](https://github.com/INN/Largo/blob/develop/inc/verify.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/verify.html)
- [ ] [./inc/widgets.php](https://github.com/INN/Largo/blob/develop/inc/widgets.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/widgets.html)
- [ ] [./inc/avatars/admin.php](https://github.com/INN/Largo/blob/develop/inc/avatars/admin.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/avatars/admin.html)
- [ ] [./inc/avatars/functions.php](https://github.com/INN/Largo/blob/develop/inc/avatars/functions.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/avatars/functions.html)
- [ ] [./inc/customizer/class-largo-wp-customize-multi-checkbox-control.php](https://github.com/INN/Largo/blob/develop/inc/customizer/class-largo-wp-customize-multi-checkbox-control.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/customizer/class-largo-wp-customize-multi-checkbox-control.html)
- [ ] [./inc/customizer/class-largo-wp-customize-rich-radio-control.php](https://github.com/INN/Largo/blob/develop/inc/customizer/class-largo-wp-customize-rich-radio-control.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/customizer/class-largo-wp-customize-rich-radio-control.html)
- [ ] [./inc/customizer/class-largo-wp-customize-textarea-control.php](https://github.com/INN/Largo/blob/develop/inc/customizer/class-largo-wp-customize-textarea-control.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/customizer/class-largo-wp-customize-textarea-control.html)
- [ ] [./inc/customizer/customizer.php](https://github.com/INN/Largo/blob/develop/inc/customizer/customizer.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/customizer/customizer.html)
- [ ] [./inc/picturefill/picturefill-wp.php](https://github.com/INN/Largo/blob/develop/inc/picturefill/picturefill-wp.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/picturefill/picturefill-wp.html)
- [ ] [./inc/picturefill/inc/class-model-picturefill-wp.php](https://github.com/INN/Largo/blob/develop/inc/picturefill/inc/class-model-picturefill-wp.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/picturefill/inc/class-model-picturefill-wp.html)
- [ ] [./inc/picturefill/inc/class-picturefill-wp.php](https://github.com/INN/Largo/blob/develop/inc/picturefill/inc/class-picturefill-wp.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/picturefill/inc/class-picturefill-wp.html)
- [ ] [./inc/picturefill/inc/class-view-picturefill-wp.php](https://github.com/INN/Largo/blob/develop/inc/picturefill/inc/class-view-picturefill-wp.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/picturefill/inc/class-view-picturefill-wp.html)
- [ ] [./inc/picturefill/inc/templates/fallback-source-template.php](https://github.com/INN/Largo/blob/develop/inc/picturefill/inc/templates/fallback-source-template.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/picturefill/inc/templates/fallback-source-template.html)
- [ ] [./inc/picturefill/inc/templates/noscript-template.php](https://github.com/INN/Largo/blob/develop/inc/picturefill/inc/templates/noscript-template.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/picturefill/inc/templates/noscript-template.html)
- [ ] [./inc/picturefill/inc/templates/picture-template.php](https://github.com/INN/Largo/blob/develop/inc/picturefill/inc/templates/picture-template.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/picturefill/inc/templates/picture-template.html)
- [ ] [./inc/picturefill/inc/templates/source-template.php](https://github.com/INN/Largo/blob/develop/inc/picturefill/inc/templates/source-template.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/picturefill/inc/templates/source-template.html)
- [ ] [./inc/widgets/largo-about.php](https://github.com/INN/Largo/blob/develop/inc/widgets/largo-about.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/widgets/largo-about.html)
- [ ] [./inc/widgets/largo-author-bio.php](https://github.com/INN/Largo/blob/develop/inc/widgets/largo-author-bio.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/widgets/largo-author-bio.html)
- [ ] [./inc/widgets/largo-disclaimer-widget.php](https://github.com/INN/Largo/blob/develop/inc/widgets/largo-disclaimer-widget.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/widgets/largo-disclaimer-widget.html)
- [ ] [./inc/widgets/largo-donate.php](https://github.com/INN/Largo/blob/develop/inc/widgets/largo-donate.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/widgets/largo-donate.html)
- [ ] [./inc/widgets/largo-explore-related.php](https://github.com/INN/Largo/blob/develop/inc/widgets/largo-explore-related.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/widgets/largo-explore-related.html)
- [ ] [./inc/widgets/largo-facebook.php](https://github.com/INN/Largo/blob/develop/inc/widgets/largo-facebook.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/widgets/largo-facebook.html)
- [ ] [./inc/widgets/largo-featured.php](https://github.com/INN/Largo/blob/develop/inc/widgets/largo-featured.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/widgets/largo-featured.html)
- [ ] [./inc/widgets/largo-follow.php](https://github.com/INN/Largo/blob/develop/inc/widgets/largo-follow.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/widgets/largo-follow.html)
- [ ] [./inc/widgets/largo-footer-featured.php](https://github.com/INN/Largo/blob/develop/inc/widgets/largo-footer-featured.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/widgets/largo-footer-featured.html)
- [ ] [./inc/widgets/largo-image-widget.php](https://github.com/INN/Largo/blob/develop/inc/widgets/largo-image-widget.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/widgets/largo-image-widget.html)
- [ ] [./inc/widgets/largo-inn-rss.php](https://github.com/INN/Largo/blob/develop/inc/widgets/largo-inn-rss.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/widgets/largo-inn-rss.html)
- [ ] [./inc/widgets/largo-post-series-links.php](https://github.com/INN/Largo/blob/develop/inc/widgets/largo-post-series-links.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/widgets/largo-post-series-links.html)
- [ ] [./inc/widgets/largo-prev-next-post-links.php](https://github.com/INN/Largo/blob/develop/inc/widgets/largo-prev-next-post-links.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/widgets/largo-prev-next-post-links.html)
- [ ] [./inc/widgets/largo-recent-comments.php](https://github.com/INN/Largo/blob/develop/inc/widgets/largo-recent-comments.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/widgets/largo-recent-comments.html)
- [ ] [./inc/widgets/largo-recent-posts.php](https://github.com/INN/Largo/blob/develop/inc/widgets/largo-recent-posts.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/widgets/largo-recent-posts.html)
- [ ] [./inc/widgets/largo-related-posts.php](https://github.com/INN/Largo/blob/develop/inc/widgets/largo-related-posts.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/widgets/largo-related-posts.html)
- [ ] [./inc/widgets/largo-series-posts.php](https://github.com/INN/Largo/blob/develop/inc/widgets/largo-series-posts.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/widgets/largo-series-posts.html)
- [ ] [./inc/widgets/largo-sidebar-featured.php](https://github.com/INN/Largo/blob/develop/inc/widgets/largo-sidebar-featured.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/widgets/largo-sidebar-featured.html)
- [ ] [./inc/widgets/largo-staff.php](https://github.com/INN/Largo/blob/develop/inc/widgets/largo-staff.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/widgets/largo-staff.html)
- [ ] [./inc/widgets/largo-tag-list.php](https://github.com/INN/Largo/blob/develop/inc/widgets/largo-tag-list.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/widgets/largo-tag-list.html)
- [ ] [./inc/widgets/largo-taxonomy-list.php](https://github.com/INN/Largo/blob/develop/inc/widgets/largo-taxonomy-list.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/widgets/largo-taxonomy-list.html)
- [ ] [./inc/widgets/largo-twitter.php](https://github.com/INN/Largo/blob/develop/inc/widgets/largo-twitter.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/widgets/largo-twitter.html)
- [ ] [./inc/wp-taxonomy-landing/taxonomy-landing.php](https://github.com/INN/Largo/blob/develop/inc/wp-taxonomy-landing/taxonomy-landing.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/wp-taxonomy-landing/taxonomy-landing.html)
- [ ] [./inc/wp-taxonomy-landing/functions/cftl-admin.php](https://github.com/INN/Largo/blob/develop/inc/wp-taxonomy-landing/functions/cftl-admin.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/wp-taxonomy-landing/functions/cftl-admin.html)
- [ ] [./inc/wp-taxonomy-landing/functions/cftl-intercept-queries.php](https://github.com/INN/Largo/blob/develop/inc/wp-taxonomy-landing/functions/cftl-intercept-queries.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/wp-taxonomy-landing/functions/cftl-intercept-queries.html)
- [ ] [./inc/wp-taxonomy-landing/functions/cftl-series-order.php](https://github.com/INN/Largo/blob/develop/inc/wp-taxonomy-landing/functions/cftl-series-order.php) - [docs](http://largo.readthedocs.org/en/develop/api/inc/wp-taxonomy-landing/functions/cftl-series-order.html)
- [ ] [./partials/author-bio-description.php](https://github.com/INN/Largo/blob/develop/partials/author-bio-description.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/author-bio-description.html)
- [ ] [./partials/author-bio-social-links.php](https://github.com/INN/Largo/blob/develop/partials/author-bio-social-links.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/author-bio-social-links.html)
- [ ] [./partials/content-argolinks.php](https://github.com/INN/Largo/blob/develop/partials/content-argolinks.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/content-argolinks.html)
- [ ] [./partials/content-not-found.php](https://github.com/INN/Largo/blob/develop/partials/content-not-found.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/content-not-found.html)
- [ ] [./partials/content-page.php](https://github.com/INN/Largo/blob/develop/partials/content-page.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/content-page.html)
- [ ] [./partials/content-series.php](https://github.com/INN/Largo/blob/develop/partials/content-series.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/content-series.html)
- [ ] [./partials/content-single-classic.php](https://github.com/INN/Largo/blob/develop/partials/content-single-classic.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/content-single-classic.html)
- [ ] [./partials/content-single.php](https://github.com/INN/Largo/blob/develop/partials/content-single.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/content-single.html)
- [ ] [./partials/content-tiny.php](https://github.com/INN/Largo/blob/develop/partials/content-tiny.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/content-tiny.html)
- [ ] [./partials/content.php](https://github.com/INN/Largo/blob/develop/partials/content.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/content.html)
- [ ] [./partials/footer-before-footer-widget-area.php](https://github.com/INN/Largo/blob/develop/partials/footer-before-footer-widget-area.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/footer-before-footer-widget-area.html)
- [ ] [./partials/footer-boilerplate.php](https://github.com/INN/Largo/blob/develop/partials/footer-boilerplate.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/footer-boilerplate.html)
- [ ] [./partials/footer-sticky.php](https://github.com/INN/Largo/blob/develop/partials/footer-sticky.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/footer-sticky.html)
- [ ] [./partials/footer-widget-area.php](https://github.com/INN/Largo/blob/develop/partials/footer-widget-area.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/footer-widget-area.html)
- [ ] [./partials/header-ad-zone.php](https://github.com/INN/Largo/blob/develop/partials/header-ad-zone.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/header-ad-zone.html)
- [ ] [./partials/home-bottom-widget-area.php](https://github.com/INN/Largo/blob/develop/partials/home-bottom-widget-area.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/home-bottom-widget-area.html)
- [ ] [./partials/home-post-list.php](https://github.com/INN/Largo/blob/develop/partials/home-post-list.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/home-post-list.html)
- [ ] [./partials/home-topstories.php](https://github.com/INN/Largo/blob/develop/partials/home-topstories.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/home-topstories.html)
- [ ] [./partials/homepage-alert.php](https://github.com/INN/Largo/blob/develop/partials/homepage-alert.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/homepage-alert.html)
- [ ] [./partials/largo-header.php](https://github.com/INN/Largo/blob/develop/partials/largo-header.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/largo-header.html)
- [ ] [./partials/nav-global.php](https://github.com/INN/Largo/blob/develop/partials/nav-global.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/nav-global.html)
- [ ] [./partials/nav-main.php](https://github.com/INN/Largo/blob/develop/partials/nav-main.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/nav-main.html)
- [ ] [./partials/nav-secondary.php](https://github.com/INN/Largo/blob/develop/partials/nav-secondary.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/nav-secondary.html)
- [ ] [./partials/nav-sticky.php](https://github.com/INN/Largo/blob/develop/partials/nav-sticky.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/nav-sticky.html)
- [ ] [./partials/sidebar-archive.php](https://github.com/INN/Largo/blob/develop/partials/sidebar-archive.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/sidebar-archive.html)
- [ ] [./partials/sidebar-fallback.php](https://github.com/INN/Largo/blob/develop/partials/sidebar-fallback.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/sidebar-fallback.html)
- [ ] [./partials/sidebar-single.php](https://github.com/INN/Largo/blob/develop/partials/sidebar-single.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/sidebar-single.html)
- [ ] [./partials/sidebar.php](https://github.com/INN/Largo/blob/develop/partials/sidebar.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/sidebar.html)
- [ ] [./partials/single-hero.php](https://github.com/INN/Largo/blob/develop/partials/single-hero.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/single-hero.html)
- [ ] [./partials/sticky-posts.php](https://github.com/INN/Largo/blob/develop/partials/sticky-posts.php) - [docs](http://largo.readthedocs.org/en/develop/api/partials/sticky-posts.html)
- [ ] [./tests/bootstrap.php](https://github.com/INN/Largo/blob/develop/tests/bootstrap.php) - [docs](http://largo.readthedocs.org/en/develop/api/tests/bootstrap.html)
- [ ] [./tests/homepages/test-homepage-class.php](https://github.com/INN/Largo/blob/develop/tests/homepages/test-homepage-class.php) - [docs](http://largo.readthedocs.org/en/develop/api/tests/homepages/test-homepage-class.html)
- [ ] [./tests/homepages/test-homepage.php](https://github.com/INN/Largo/blob/develop/tests/homepages/test-homepage.php) - [docs](http://largo.readthedocs.org/en/develop/api/tests/homepages/test-homepage.html)
- [ ] [./tests/inc/test-ajax-functions.php](https://github.com/INN/Largo/blob/develop/tests/inc/test-ajax-functions.php) - [docs](http://largo.readthedocs.org/en/develop/api/tests/inc/test-ajax-functions.html)
- [ ] [./tests/inc/test-avatars.php](https://github.com/INN/Largo/blob/develop/tests/inc/test-avatars.php) - [docs](http://largo.readthedocs.org/en/develop/api/tests/inc/test-avatars.html)
- [ ] [./tests/inc/test-enqueue.php](https://github.com/INN/Largo/blob/develop/tests/inc/test-enqueue.php) - [docs](http://largo.readthedocs.org/en/develop/api/tests/inc/test-enqueue.html)
- [ ] [./tests/inc/test-featured-content.php](https://github.com/INN/Largo/blob/develop/tests/inc/test-featured-content.php) - [docs](http://largo.readthedocs.org/en/develop/api/tests/inc/test-featured-content.html)
- [ ] [./tests/inc/test-featured-media.php](https://github.com/INN/Largo/blob/develop/tests/inc/test-featured-media.php) - [docs](http://largo.readthedocs.org/en/develop/api/tests/inc/test-featured-media.html)
- [ ] [./tests/inc/test-helpers.php](https://github.com/INN/Largo/blob/develop/tests/inc/test-helpers.php) - [docs](http://largo.readthedocs.org/en/develop/api/tests/inc/test-helpers.html)
- [ ] [./tests/inc/test-post-tags.php](https://github.com/INN/Largo/blob/develop/tests/inc/test-post-tags.php) - [docs](http://largo.readthedocs.org/en/develop/api/tests/inc/test-post-tags.html)
- [ ] [./tests/inc/test-post-templates.php](https://github.com/INN/Largo/blob/develop/tests/inc/test-post-templates.php) - [docs](http://largo.readthedocs.org/en/develop/api/tests/inc/test-post-templates.html)
- [ ] [./tests/inc/test-taxonomies.php](https://github.com/INN/Largo/blob/develop/tests/inc/test-taxonomies.php) - [docs](http://largo.readthedocs.org/en/develop/api/tests/inc/test-taxonomies.html)
- [ ] [./tests/inc/test-update.php](https://github.com/INN/Largo/blob/develop/tests/inc/test-update.php) - [docs](http://largo.readthedocs.org/en/develop/api/tests/inc/test-update.html)
- [ ] [./tests/inc/test-users.php](https://github.com/INN/Largo/blob/develop/tests/inc/test-users.php) - [docs](http://largo.readthedocs.org/en/develop/api/tests/inc/test-users.html)
- [ ] [./tests/inc/widgets/test-disclaimer-widget.php](https://github.com/INN/Largo/blob/develop/tests/inc/widgets/test-disclaimer-widget.php) - [docs](http://largo.readthedocs.org/en/develop/api/tests/inc/widgets/test-disclaimer-widget.html)
- [ ] [./tests/inc/widgets/test-largo-post-series-links.php](https://github.com/INN/Largo/blob/develop/tests/inc/widgets/test-largo-post-series-links.php) - [docs](http://largo.readthedocs.org/en/develop/api/tests/inc/widgets/test-largo-post-series-links.html)
- [ ] [./tests/inc/widgets/test-largo-series-posts.php](https://github.com/INN/Largo/blob/develop/tests/inc/widgets/test-largo-series-posts.php) - [docs](http://largo.readthedocs.org/en/develop/api/tests/inc/widgets/test-largo-series-posts.html)
- [ ] [./tests/mock/mock-admin-functions.php](https://github.com/INN/Largo/blob/develop/tests/mock/mock-admin-functions.php) - [docs](http://largo.readthedocs.org/en/develop/api/tests/mock/mock-admin-functions.html)
- [ ] [./tests/mock/mock-options-framework.php](https://github.com/INN/Largo/blob/develop/tests/mock/mock-options-framework.php) - [docs](http://largo.readthedocs.org/en/develop/api/tests/mock/mock-options-framework.html)
- [ ] [./tests/mock/test-mock-admin-functions.php](https://github.com/INN/Largo/blob/develop/tests/mock/test-mock-admin-functions.php) - [docs](http://largo.readthedocs.org/en/develop/api/tests/mock/test-mock-admin-functions.html)
- [ ] [./tests/templates/test-homepage.php](https://github.com/INN/Largo/blob/develop/tests/templates/test-homepage.php) - [docs](http://largo.readthedocs.org/en/develop/api/tests/templates/test-homepage.html)

## Will Not Document

**Comment by tothebeat** [comment]
Individual tickets have been created for each of the PHP files in Largo.

I've added a section to our [WordPress style guide on inline documentation](https://github.com/INN/docs/blob/master/style-guides/code/wordpress.md#inline-documentation), which includes a recommendation to include a file header DocBlock in each file. With this, every PHP file should have at least that added.

The `Makefile` now uses a Python script to find the PHP files of Largo and extract documentation to RST, so there is no longer an issue of needing to manually add/remove files from the `Makefile` itself.

[doxphp](https://github.com/avalanche123/doxphp) may not be the tool we ultimately use to extract the inline documentation, for the fact that it picks up things other than DocBlocks (properties being extracted automatically from [archive.php](http://largo.readthedocs.org/en/develop/api/archive.html)) and has no real documentation of its own making customization difficult.

**Comment by tothebeat** [comment]
Not complete but tracked in other tickets.

**Comment by rclations** [comment]
In an effort to clean up the issue queue and focus our energy in one direction, I'm going to close this  issue in favor of #1449, which should take care of everything covered in this issue (and child issues)
